monai/transforms/rand_rotation.py

```diff
@@ -0,0 +1,158 @@
+        self._rand_k = np.random.randint(self.max_k) + 1
+        return self
+
+    def __call__(self, img):
+        if not self._do_transform:
+            return img
+        return Rotate90(self._rand_k, self.axes)(img)
+
+
+@export
+def fn_map(transform, map_key=None, common_key=None, inplace=True, randomize=OFF):
+    """
+    Convert a vanilla `transform` into a dictionary-based transform.
+
+    The new transform takes a dictionary as input and returns a dictionary.
+    each data[map_key] of the transform's input will be transformed by `transform`,
+    with all data[common_key] as additional arguments.
+
+    Args:
+        transform (callable): a transform to be wrapped into a dict-based transform.
+        map_key (str or list/tuple of str): keys to apply `transform`.
+        common_key (str or list/tuple of str): keys of additional arguments to be passed to `transform` from data dict.
+        inplace (bool): whether to modify the data dict inplace.
+        randomize (SYNC|ASYNC|OFF):
+            SYNC: transform will be randomized once and applied to each of `map_key`.
+            ASYNC: transform applied to each `map_key` will be randomized beforehand.
+            OFF: no randomization for the transform.
+    """
+    if map_key:
+        _map_key = map_key if isinstance(map_key, (list, tuple)) else (map_key,)
+    else:
+        _map_key = None
+    if common_key:
+        _common_key = common_key if isinstance(common_key, (list, tuple)) else (common_key,)
+    else:
+        _common_key = None
+    if randomize not in (SYNC, ASYNC, OFF):
+        raise ValueError('Invalid option {} for `randomize`'.format(randomize))
+
+    if not isinstance(transform, Randomizable):
+        if randomize != OFF:
+            raise ValueError('transform {} not randomizable'.format(transform))
+    _transform = copy.deepcopy(transform)
+
+    def _dict_transform(data):
+        if not isinstance(data, MutableMapping):
+            raise ValueError('{} must be a dictionary'.format(type(data)))
+        d = copy.copy(data) if not inplace else data
+
+        common_args = {name: d.get(name, None) for name in _common_key} if _common_key else None
+        apply_key = _map_key or tuple(d)
+        if randomize in (SYNC, ASYNC):
+            _transform.randomize()
+        for key in apply_key:
+            if common_args:
+                d[key] = _transform(d[key], **common_args)
+            else:
+                d[key] = _transform(d[key])
+            if randomize == ASYNC:
+                _transform.randomize()
+        return d
+
+    return _dict_transform
+
+
+if __name__ == "__main__":
+    data = {
+        'img': np.array((1, 2, 3, 4)).reshape((1, 2, 2)),
+        'seg': np.array((1, 2, 3, 4)).reshape((1, 2, 2)),
+        'affine': 3,
+        'dtype': 4,
+        'unused': 5,
+    }
+    rotator = RandRotate90(0.8)
+    data_result = fn_map(rotator, map_key=['img', 'seg'], randomize=SYNC)(data)
```

Comment:
I think you want to use "fn_map" to cover all kinds of dict->numpy mapping.
Can we still make a transforms chain for the dict data?
If you use 2 levels transforms directly, you can get some logic like:
```py
transforms = [
    Crop(keys=['img', 'label'], 0.8),
    Noise(keys=['img'], 0.3),
    IntensityNormalize(keys=['img'])
]
for t in transforms:
    data = t(data)
```

Reply:
my hope here is that `fn_map(.., map_key=['img', 'seg'])` takes care of all the dictionary handling logic, so that `Crop` `Noise` `IntensityNormalize` are "vanilla" transforms that deal with numpy array/torch tensors directly.

Also the users could use `Crop(0.8)` `Noise(0.3)` directly without using the wrapper.

Reply:
a transform chain will be something like:
```
transforms = [
    fn_map(Crop(0.8), map_key=['img', 'seg'], randomize=SYNC),
    fn_map(Noise(0.3), map_key=['img']),
    fn_map(IntensityNormalize(), map_key=['img'])),
]
for t in transforms:
    data = t(data)
```